docs/HACKING.md

```diff
@@ -160,3 +159,19 @@ will attempt to clone them to a temporary directory.
 Bumping the release and manifest versions typically require updating the `elasticsearch-operator` as well.
 * `dep ensure -update github.com/openshift/elasticsearch-operator`
 
+
+# Debugging tips
+
+## Where to start when deployment times out?
+
+Where to start looking when this happends:
+```
+make deploy
+...
+waiting the deployment openshift-logging/cluster-logging-operator - remaining attempts: 0
+```
+
+* `oc project openshift-logging` following commands assume this. 
+* `oc get all` missing/unexpected resources?
+* `oc describe all` check conditions and events.
+

```

Comment:
TBH, I came up with a more concrete approach to debug OLM deployments based on the chain of events and CRs created and reconciled by the two OLM operators namely the `catalog-operator` and the `olm-operator`. The resources collected below are part of our `logging.dump.sh` [script](https://github.com/openshift/origin-aggregated-logging/blob/master/hack/logging-dump.sh).

In general the OLM operator installation process works like this (explained more visually in our Openshift docs though):
1. Operator installation starts by creating a **subscription** for the registry's catalog. First hint to check for broken requirements is : `oc get subscription -o yaml`
2. The **subscription** is translated in an **install plan** that tells OLM what resources to install, i.e. CSV, RBAC, CRDs, ServiceAccount. So the next hint is: `oc get installplan -o yaml`
3. If the **install plan** was applied correctly our should see the installed resources, ie. **CSV, rbac, CRDs, ServiceAccount, Deployment** in your target namespace: So next hint is: `oc get csv -o yaml`
4. If the pods don't start, you should check the logs of your operator, e.g `oc logs cluster-logging-operator-...`
5. Finally, checking the logs of the operator-framework can also give hints on what is broken:
5.1 Logs for **catalog-operator**: `oc -n openshift-operator-lifecycle-manager logs catalog-operator-...`
5.2 Logs for **olm-operator**: `oc -n openshift-operator-lifecycle-manager logs olm-operator-...`

Reply:
All this stuff taken out of this PR, will do a separate PR to improve the HACKING doc.